vyper/ast/parse.py

```diff
@@ -82,7 +83,7 @@ def parse_to_ast_with_settings(
     module = vy_ast.get_node(py_ast)
     assert isinstance(module, vy_ast.Module)  # mypy hint
 
-    return settings, module
+    return settings, module, loop_var_annotations
```

Comment:
i think instead of passing around `loop_var_annotations` everywhere we should just tag it on the AST during the `annotate_vyper_ast` portion of ast parsing.